src/set_theory/principal.lean

```diff
@@ -67,4 +69,37 @@ theorem nfp_le_of_principal {op : ordinal → ordinal → ordinal}
   {a o : ordinal} (hao : a < o) (ho : principal op o) : nfp (op a) a ≤ o :=
 nfp_le.2 $ λ n, le_of_lt (ho.iterate_lt hao n)
 
+/-! ### Principal ordinals are unbounded -/
+
+/-- The least strict upper bound of `op` applied to all pairs of ordinals less than `a`. -/
+def lsub_op_lt (op : ordinal → ordinal → ordinal) (o : ordinal) : ordinal :=
```

Comment:
I'm not a fan of this name.
This is roughly a binary version of `ordinal.blsub`, right? Maybe `ordinal.blsub2`?

Reply:
I like that name and I've changed it accordingly.